File: mobile/lib/features/service_requests/models/service_request_attachment.dart
class ServiceRequestAttachment {
  const ServiceRequestAttachment({
    required this.id,
    required this.slot,
    required this.contentType,
    required this.fileSizeBytes,
    required this.width,
    required this.height,
    required this.createdAt,
  });
  final String id;
  final int slot;
  final String contentType;
  final int fileSizeBytes;
  final int width;
  final int height;
  final DateTime createdAt;

  factory ServiceRequestAttachment.fromJson(Map<String, dynamic> json) =>
      ServiceRequestAttachment(
        id: json['id'] as String,
        slot: json['slot'] as int,
        contentType: json['contentType'] as String,
        fileSizeBytes: json['fileSizeBytes'] as int,
        width: json['width'] as int,
        height: json['height'] as int,
        createdAt: DateTime.parse(json['createdAt'] as String),
      );
}

File: backend/src/AssistLK.Domain/Entities/ServiceRequestAttachment.cs
namespace AssistLK.Domain.Entities;

public class ServiceRequestAttachment : BaseEntity
{
    public Guid ServiceRequestId { get; set; }
    public ServiceRequest ServiceRequest { get; set; } = null!;
    public string StorageKey { get; set; } = string.Empty;
    public string ContentType { get; set; } = "image/jpeg";
    public long FileSizeBytes { get; set; }
    public int Width { get; set; }
    public int Height { get; set; }
    public int Slot { get; set; }
    public string ContentHash { get; set; } = string.Empty;
}
